pwlf/pwlf.py

```diff
@@ -189,39 +188,29 @@ def __init__(self, x, y, disp_res=False, lapack_driver='gelsd', degree=1,
 
         self.print = disp_res
         self.lapack_driver = lapack_driver
-        # x and y should be numpy arrays
-        # if they are not convert to numpy array
-        if isinstance(x, np.ndarray) is False:
-            x = np.array(x)
-        if isinstance(y, np.ndarray) is False:
-            y = np.array(y)
+        x, y = self._switch_to_np_array(x), self._switch_to_np_array(y)
 
-        self.x_data = x
-        self.y_data = y
+        self.x_data, self.y_data = x, y
 
         # calculate the number of data points
         self.n_data = x.size
 
         # set the first and last break x values to be the min and max of x
-        self.break_0 = np.min(self.x_data)
-        self.break_n = np.max(self.x_data)
+        self.break_0, self.break_n = np.min(self.x_data), np.max(self.x_data)
 
-        if degree < 12 and degree >= 0:
+        if 12 > degree >= 0:
             # I actually don't know what the upper degree limit should
             self.degree = int(degree)
         else:
-            not_suported = "degree = " + str(degree) + " is not supported."
-            raise ValueError(not_suported)
+            raise ValueError(f"degree = {degree} is not supported.")
```

Comment:
So these f-Strings are great! Unfortunately, they are not supported until Python```>=3.6```... Let me think a bit if I want to drop Python 3.5 support.